plugins/kubernetes/src/api/KubernetesBackendClient.ts

```diff
@@ -110,4 +131,32 @@ export class KubernetesBackendClient implements KubernetesApi {
 
     return (await this.handleResponse(response)).items;
   }
+
+  async proxy(options: {
+    clusterName: string;
+    path: string;
+    init?: RequestInit;
+  }): Promise<Response> {
+    const { authProvider } = await this.getCluster(options.clusterName);
+    const k8sToken = await this.getBearerToken(authProvider);
+    const url = `${await this.discoveryApi.getBaseUrl('kubernetes')}/proxy${
+      options.path
+    }`;
+    const identityResponse = await this.identityApi.getCredentials();
+    const headers = identityResponse.token
+      ? {
+          ...options.init?.headers,
+          [`Backstage-Kubernetes-Cluster`]: options.clusterName,
+          [`Backstage-Kubernetes-Authorization`]: `Bearer ${k8sToken}`,
+          Authorization: `Bearer ${identityResponse.token}`,
+        }
+      : {
+          ...options.init?.headers,
+          [`Backstage-Kubernetes-Cluster`]: options.clusterName,
+          [`Backstage-Kubernetes-Authorization`]: `Bearer ${k8sToken}`,
+        };
```

Comment:
I think what @Rugvip means is
```suggestion
    const headers = {
      ...options.init?.headers,
      [`Backstage-Kubernetes-Cluster`]: options.clusterName,
      [`Backstage-Kubernetes-Authorization`]: `Bearer ${k8sToken}`,
      ...(identityResponse.token && {Authorization: `Bearer ${identityResponse.token}`})
    };
```

Reply:
Thanks this helps, i was exploring how that expression gets evaluated because its the first time ive seen it. Pretty neat trick. Refactored conditional.